General Definition Cleanup - 2/12/2025

robertgk2017 opened the pull request:
TES3 - Enum/Flags Indentation and some Name updates
TES3 - More/Readable common variables
TES3 \ wbEffects - Add DontShows and a Range Union to properly replicate the Construction Sets restrictions
TES3 \ wbGlobalAfterLoad -  Force large negative value check to a Single
TES3 \ wbModel - Set name to `Model`
TES3 \ ENCH \ ENDT - Update Auto Calculate flag
TES3 \ GLOB - Remove FLTV Union, its all the same thing.
TES3 \ INFO \ Conditions - Rebuilt conditions from scratch, and added a Union to separate out the 3rd/4th bytes according to the CS.
TES3 \ LIGH - Set Default Values, Set Required Sub-Records, and Update Radius Def Type
TES3 \ LOCK - Set Default Values and Required Sub-Records
TES3 \ LTEX - Set Required Sub-Records
TES3 \ MGEF - Update struct name from Color to Lighting Effect to match CK.
TES3 \ MISC - Set Default Values and Required Sub-Records
TES3 \ NPC - Set Default Values and Required Fields
TES3 \ PROB - Set Default Values and Required Sub-Records
TES3 \ RACE - Set Default Values and Required Sub-Records
TES3 \ REGN - Set Default Values and Required Sub-Records
TES3 \ REPA - Set default values and Required sub-records, made Value and Uses Unsigned Ints
TES3 \ SOUN - Set required Sub-Records and Default Value
TES3 \ WEAP - Set default values

TES4 \ ARMO - Group Model Sub-Records
TES4 \ CLOT - Group Model Sub-Records
TES4 \ CREA \ NIFZ - Change name to Model List to make CK

TES4R \ CELL - New XLRL and XTLI subrecords
TES4R \ REFR - New XAAG and XACN subrecords

FO3 \ ARMO - Group Model Sub-Records
FO3 \ ARMA - Group Model Sub-Records

FNV \ ARMO - Group Model Sub-Records
FNV \ARMA - Group Model Sub-Records
FNV \ WEAP \ DNAM \ Reload Animation - Update Enum/AfterLoad to set any instance of 255 to 0

TES5 \ wbPerkDATAQuestStageToStr - Fix broken quest record targeting
TES5 \ wbTintLayerToStr - Update Race Tint element paths
TES5 \ ARMO - Group Model Sub-Records
TES5 \ ARMA - Group Model Sub-Records
TES5 \ RACE - Update Tint Masks array

TES5/FO4/FO76 \ wbConditions - remove dfNotAlignable

TES5/FO4/FO76/SF1 \ NPC_ - Remove (Actor) from the record name, because why.

FO4 \ ARMO - Group Model Sub-Records
FO4 \ ARMA - Group Model Sub-Records
FO4 \ IDLM - Swap Model and QNAM sub-record order
FO4 \ IDLM - Add PNAM/QNAM mutually exclusive sub-records with AfterSets and DontShows
FO4 \ INGR - Add PTRN Sub-Record
FO4 \ LIGH - Swap LNAM and SNAM sub-record order
FO4 \ REFR - Add XPWR Sub-Record
FO4 \ ReferenceRecord - Add wbRagdoll and XHLT

FO76 \ ARMO - Group Model Sub-Records
FO76 \ ARMA - Group Model Sub-Records
FO76 \ PLYT - Add ANAM/BNAM names
FO76 \ wbObjectTemplate - fix field name

SF1 \ ARMA - Group Model Sub-Records
SF1 \ COBJ \ CIFK - Decoded as Instanced Filter Keyword
SF1 \ INFO - Add TPIC, same as previous games.  Credit to Jon
SF1 \ WRLD \ CNAM - Exclude from being set when no Parent Worldspace is set.

Common \ wbMenuModeEnum - Update names/IsGame checks
Common \ wbPlacedAddInfo - Add check for Cell being Assigned
Common \ wbWeatherTimeOfDay - Updated, now with 100% less jank!
Common \ wbWorldMaxHeightData - Exclude FO4 from dfNoCopyAsOverride

DataFormatNif \ TwbNifFile.UpdateNifVersion  - Uncomment version check for v10010, and added v1010101

Implementation \ TwbContainedInElement.DoAfterSet - Add QUST Children GRUP Handling for FO4+

Script - add gmTES3 as a recognized wbGameMode
Script \ Assets Manager - Make Papyrus Source a seperate asset type, and update GUI
Script \ Assets Manager - Add TES3 Support
Script \ Assets Manager - Overhaul TES4 Support
Script \ Assets Manager - Update ARMO/ARMA Model handling
Script \ Assets Manager \ TES5 - Add RACE Tint Mask Textures Support
Script \ Check For Invalid Alternate Textures - Update ARMO/ARMA Model Handling
Script \ Export Dialogues - Now properly considers Quest Conditions when calculating voice types - Thanks Nighthawk!
Script \ Export Dialogues - Update 2 Element Paths to reflect recent definition changes
Script \ Find ESP Plugins which could be turned into ESL - fix missing paramter on line 72
Script \ List Used Meshes From References - Update ARMO/ARMA Model Handling
Script \ Replace Model File Name - Update ARMO/ARMA Model Handling

xeMainForm - Add Yes To All functionality to Change Form Ids
xeMainForm - Fix Change Form IDs reporting wrong number of processed records

xEdit - change all uses of the name Overlay for Update in relation to the Starfield File Flag
xEdit - Added Oblivion Remastered support

robertgk2017 committed:
Common \ wbPlacedAddInfo - add check for Cell being Assigned to the reference in question

robertgk2017 committed:
TES5 \ wbPerkDATAQuestStageToStr - Fix broken quest record targeting

jbostrus committed:
Feature: Add option in filter dialog for element contains text.
* Also supported for scripted filter. Use FilterByElementValue and FilterElementValue.

robertgk2017 committed:
Common - Update wbMenuModeEnum

robertgk2017 committed:
TES5/FO4/FO76 \ wbConditions - remove dfNotAlignable

robertgk2017 committed:
SF1 \ INFO - Add TPIC, same as previous games.  Credit to Jon

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ NPC_ - Remove (Actor) from the record name, because why.

jbostrus committed:
SF1: Added EFRM and EORI which are present on Surface
Component DAT2-Block Height Adjustments added names for entities.
Component DAT2-Surface Tree Pattern Swap partiall decoded.
Component DATA-Overlay Designated Placement Info decoded.
Renamed element headers for Rows/Columns which were backwards and mislabeled in cases. This will impact scripts.

robertgk2017 committed:
SF1 \ WTHS - Add wbConditions

robertgk2017 committed:
TES3 - Enum/Flags Indentation and some Name updates

robertgk2017 committed:
TES3 - More/Readable common variables

robertgk2017 committed:
TES3 \ WEAP - Set default values

robertgk2017 committed:
TES3 \ wbEffects - Add DontShows and a Range Union to properly replicate the Construction Sets restrictions

robertgk2017 committed:
TES3 \ SOUN - Set required Sub-Records and Default Value

robertgk2017 committed:
TES3 \ GLOB - Remove FLTV Union, its all the same thing.

robertgk2017 committed:
TES3 - Misc Flags/Counters/etc

robertgk2017 committed:
TES3 \ wbEffects - DontShow typo fix + one more AfterLoad to finish it out

robertgk2017 committed:
TES3 \ ENCH \ ENDT - Update Auto Calculate flag

robertgk2017 committed:
TES3 \ INFO \ Conditions - Rebuilt conditions from scratch, and added a Union to seperate out the 3rd/4th bytes according to the CS.

robertgk2017 committed:
TES3 \ INFO \ Conditions - OOPS

robertgk2017 committed:
Script \ Find ESP Plugins which could be turned into ESL - fix missing paramter on line 72

robertgk2017 committed:
oops left a comment in

robertgk2017 committed:
TES3 \ REPA - Set default values and Required sub-records, made Value and Uses Unsigned Ints

robertgk2017 committed:
TES3 \ REGN - Set Default Values and Required Sub-Records

robertgk2017 committed:
TES3 \ RACE - Set Default Values and Required Sub-Records

robertgk2017 committed:
TES3 \ PROB - Set Default Values and Required Sub-Records

robertgk2017 committed:
Common \ wbWeatherTimeOfDay - Updated, now with 100% less jank!

robertgk2017 committed:
TES3 \ NPC - Set Default Values and Required Fields

robertgk2017 committed:
Merge branch 'pr/1415' into Misc

robertgk2017 committed:
TES3 \ MISC - Set Default Values and Required Sub-Records

robertgk2017 committed:
xeMainForm - Fix Change Form IDs reporting wrong number of processed records

robertgk2017 committed:
TES3 \ MGEF - Update struct name from Color to Lighting Effect to match CK.

robertgk2017 committed:
TES3 \ LTEX - Set Required Sub-Records

robertgk2017 committed:
TES3 \ LOCK - Set Default Values and Required Sub-Records

robertgk2017 committed:
TES3 \ LIGH - Set Default Values, Set Required Sub-Records, and Update Radius Def Type

robertgk2017 committed:
Script \ Assets Manager - Make Papyrus Source a seperate asset type, and some organization

robertgk2017 committed:
TES3 \ wbModel - Set name to `Model`

robertgk2017 committed:
Script \ Assets Manager - Update GUI

Kuroitsune commented:
Would you like to have my other work I hadn't pushed before my hand surgery since you are on a TES3 binge? You've covered most of it but I had a few extra tidbits. Mostly the String Forward fix and skill decider, a few other minor things.

robertgk2017 commented:
Late response is late.  Yea toss a PR in with your extra stuff, i'll take a look through it.

Kuroitsune committed:
*Intense hand related pain.*

Reupload for Robert to sift through for anything useful. It's probably all trash. There is a bunch of overlap. Might periodically update as my hand allows. Follow up surgery makes it hard to type.

TES3
-DESC is now displayed as the name for MGEFs and SKILs.
-NAME is now displayed as the name for SCCRs.
-Changed GridCellToFormID to wbGridCellToFormID to match the naming convention.
-Replaced the 'Record Flag' definition with an empty one. Not every Form Type used all the flags.
-Spacing adjustments. *1
-Fixed grammar.
-DELE now says 'True' instead of 'Deleted.'
-Added more common definitions. DESC, Effects, ITEX, Items, Packages, ans Spells.
-Added SetRequired on everything that was added to new Form Types by default in the CS. *2
-All mentions of 'Service Flags' are now 'Buy/Sell Service Flags.' *1
-Added SetDefaultNativeValue to everything that had a non-zero value when initialized in the CS. *2
-Moved 'Biped Objects' enumerator to it's only use instance.
-Added SetSummaryKey on many applicable definition structs to match the other game definitions.
-Added Sort Key Structs to applicable structs.
-Added Sorted Arrays to applicable arrays.
-Added AfterSet to fix forwarding bug in wbString. *3
-Replaced various uses of wbVec3 for Position/Rotation to just wbVec3PosRot when both were present.
-Each form type's 'Record Flags' defines the extra flags specific to that Form Type. *2
-Changed many instances of itS32 to itU32 when they were used for value, health, stats, etc. *4
-All mentions of 'Enchanting Charge' are now 'Enchantment Charge.'
-Added any missing wbCollapseFlags for BitMask based definitions.

\TES3 - Added AfterLoad to set the ESM flag on the record header. *5
\TES3\MAST - Renamed to 'Master Filename.'
\ALCH - Reverted name back to 'Potion.'
\ALCH\ALDT - Renamed to just 'Data' and 'Value' is now 'Potion Value.'
\APPA - Reverted name back to 'Alchemical Apparatus.'
\BODY\BYDT - Changed 'Part' to 'Body Part.'
\BODY - Set Summary Key to 'Model.'
\BOOK\BKDT - Renamed to just 'Data,' and 'Is Scroll' is just 'Scroll.'
\BOOK\TEXT - Updated to include cpTranslate.
\BSGN\FNAM - SetRequired False. *6
\BSGN\TNAM - Is now 'Constellation Filename.'
\CELL\DATA - 'Grid' is now hidden when viewing Interior Cells. *7
\CELL\NAME - Renamed to 'Location.'
\CELL - Added AfterLoad to move the value in INTV to WHGT.
\CELL - Added SetDontShow to INTV, NAM5, WHGT, and AMBI based on the 'Is Interior' bit.
\DIAL\DATA - Added Union to account for the different version (1.2 and 1.3) types. *8
\DIAL - Set Summary Key to 'Dialog Type.'
\DOOR\FNAM - SetRequired False. *6
\ENCH\ENDT - Updated the enumerator for 'Auto Calculate.' *9
\FACT\FADT - Completely restructured and renamed the definition to be more accurate and readable.
\FACT - Added ToStr to Faction Reactions for collapsed summary. *10
\GLOB\FNAM - Removed Global Union.
\GLOB - Added AfterLoad to account for several baked in errors in the Morrowind.esm. *11
\GLOB - Set Summary Key to FLTV.
\GMST - Changed names on the 'Value' signatures to be less awkward looking.
\GMST - Set Summary Key to 'Value.'
\INFO\DATA - Changed 'Gender' to 'Sex' to match the Construction Set and other games. *12
\INFO\ONAM - Updated comment for valid assignable form types.
\INFO\FNAM - Changed name to be less stilted.
\INFO\ANAM - Changed name to be less stilted.
\INFO\SCVR - Redid the entire definition. *13
\INFO - Moved 'Quest Data' struct to between 'Conditions' and BNAM.' *14
\INGR\IRDT - Vastly improved the definition with the use of an AfterLoad. *15
\LAND\VNML - Updated the definition to match the other game definitions.
\LAND\WNAM - Changed to 'World Map Colors.'
\LAND\VCLR - Now uses a special wbByteColors.
\LEVC\DATA - Renamed to simply 'Flags.'
\LEVC - Added SetCountPath for INDX.
\LEVI\DATA - Renamed to simply 'Flags.'
\LEVI - Added SetCountPath for INDX.
\LIGH\LHDT - 'Radius' is an Integer, not a float.
\LIGH\SNAM - Renamed to 'Looping Sound.'
\LTEX - Set Summary Key to 'Texture Filename.'
\MGEF\INDX - Updated the Magicka specific MGEF in the enumerator to actually say 'Magicka.' *16
\MGEF\MEDT - Removed runtime only flags from the list.
\MGEF - Added special wbByteColors for 'Color.'
\NPC_\NPDT - Optimized the union to run off the signature and not a struct.
\PGRD\NAME - Renamed to just 'Location.'
\PGRD\PGRP - Renamed to just 'Points' to match TES4 and added special wbVec3Pos for 'Position.'
\PRGD\PGRC - Renamed to 'Point-to-Point Connections' to match TES4.
\REFR - Renamed to 'Placed Object.'
\REFR\CNDT - Renamed to 'New Cell Owner.' *7
\REFR\UNAM - Now says 'True' instead of 'Blocked.'
\REFR\INTV - Renamed to 'Health/Uses Left.'
\REFR\DODT - Struct renamed to 'Teleport Destination' and the signature to Position/Rotation.
\REFR - Added AfterLoad to set, if present, the value of DELE to 0. *17
\REGN\CNAM - Renamed to 'Region Map Color.'
\SCPT\SCHD - 'Script Data Size' is now 'Compiled Size'
\SCPT\SCDT - Now show the number of lines in the script for the record Name. Thank you Zilav.
\SKIL\SKDT - 'Attribute' is now 'Governing Attribute' and added a Skill decider for 'Actions.' *18
\SOUN\DATA - Added a parameter to wbDiv to limit the number of decimals the result has. *19
\SSCR - Added AfterLoad to set, if present, the value of DELE to 0. *17

TES4
-Started spacing adjustments. *1
-Added None(6) to Magic School enumerator for Oblivion Magic Extender.
-Started fixing grammar.
-Left note about adding a union for Effects\...\Visual Effect Code. It can be either 0 or 4 bytes.
-Started decoding bits for Effects\...\Override Flags & Flags. *20
-Changed Effects\...\Unknown to a wbByteArray with 16 bytes.
\TES4\OFST - Decoded as Offset Load Order. It is entirely unused and set to be ignored.
\TES4\DELE - Tentatively decoded as Version Control. It appears unused and set to be ignored. *21
\APPA\DATA - Added decimal limit to 'Weight' and 'Quality.'
\BSGN\ICON - Changed to 'Constellation Filename.'
\CLAS\ICON - Changed to 'Image Filename'
\CREA - Bit 19 for 'Record Flags' is 'Spawns Dead.'
\CSTY\CSTD - Restructured the definition to lump as much together as possible to be cleaner looking.
\CSTY\CSAD - Restructured the definition to lump as much together as possible to be cleaner looking.
\DIAL\QSTI - Replaced with a common definition 'Associated Quests'
\DIAL\QSTR - Decoded as Removed Quests. *22
\EFSH\DATA - Restructured the definition to lump as much together as possible to be cleaner looking.
\HAIR\MODL - Changed name to 'Model.'
\MGEF\EDID - Added size to the definition. *23
\MGEF\OBME - Added note about using a union based on the handler.
\MGEF\DATA - Updated 'Unused' in 'Associated Item' to a wbByteArray 'Param A' with 4 bytes. *24
\NPC_ - Bit 19 for 'Record Flags' is 'Spawns Dead.'
\REFR\XCNT - Changed to an unsigned integer.

FO3
\DIAL\QSTI - Replaced with a common definition 'Associated Quests'
\DIAL\QSTR - Replaced with a common definition 'Removed Quests'

SF1
-Fixed a spelling error in Package Flags.

Other
-Removed unnecessary condition in an if statement related to pulling REFR out of CELL in Morrowind. (Line: 10195)
-Fixed a massive error I made when accounting for MVRF. An if statement incorrectly looked for 'MVRF' when it didn't need to, resulting in xEdit moving all REFR in a CELL with an MVRF to the temporary group. (Line: 5864)
-I discovered a huge error for TES3 that has been around since before I started decoding it. INFOs can share the same INAM value, but across different DIAL. When you load just the Morrowind.esm and do a conflict filter you'll see them. It looks like mostly copy pasted INFOs with slight changes which might explain why they share the same INAM. The INFOs with shared INAMs appear as separate INFOs in the Construction Set. None have appeared in the same DIAL.

Notes
*1 Looks like Robert and I had the same idea but did it differently. For wbServiceFlags, I updated the Apparatus to Alchemical Apparatus and added Repair for Morrowind. Spacing changes were also made to wbStringMGEFCode.

*2 I didn't think to keep any of the files I created for testing for others to view.

*3 When used inside a member definition or for a summary key, xEdit fails to forward a sized string leaving all the junk found after the terminating zero. This is a stop gap because I don't know enough to mess with that bit of xEdit.

*4 I originally went with itS32 because the Construction Set only allowed values between 0 and the signed maximum. In hindsight it made no sense to do that because negative values are not allowed. This was a quick edit after seeing Robert had done the same. May have missed one or two.

*5 Morrowind saves the file flags in the HEDR of TES3. This simply makes <ESM> appear for cosmetics.

*6 If this is not explicitly set, BSGN and DOOR with an empty/no FNAM fail to display correctly. I have no clue why, it drove me nuts. This was my solution.

*7 Never got around to grouping Interior and Exterior Cells into their own groups, but that requires more interface/implementation editing than I wanted to do at the time. I had done it for MVRF but removed those changes because I misunderstood why MVRF were moved in the CELL's data. They have to be persistent or else the game will not display them in the correct location. MVRF aren't actually moved to the new CELL they now physically exist in, so CNDT just points to the new CELL to group it into the correct CELL for the Construction Set.

*8 Should probably be reverted since it was fine before.

*9 Noticed Robert updated it too but differently. Are constant effects always auto calculated even though the Construction Set says No?

*10 I tried to modify wbFactionRelationToStr to handle Morrowind's format but it didn't work so I reverted the changes to come back to later. I did change instances of Rank to Relation because it annoyed me.

*11 In the Construction Set, it shows 0 for the affected records and they only appear on short based GLOB.

*12 Updated wbSexEnum so it now only displays 'None' for Morrowind and not in every game.

*13 I had noticed the values for what type of function in the comments I had left the first time and redid it. I completed it based on Robert's PR. Its not as in depth as his and done a bit differently but works for what I wanted originally. Removed comments not related to valid assignable form types.

*14 Some how the folks at Tamerial Rebuilt managed to get functions on Journals, which, as far as I was aware, is not possible in the Construction Set. But the Construction Set loads them just fine and saves them the same way. Also, the signatures in the struct now say 'True' instead of 'Yes.'

*15 The Construction Set sometimes set the right value (-1), other times not (0). It seemed random. So the AfterSet Changes the appropriate fields depending on the MGEF for the effect entry.

*16 Shamelessly taken from Robert's PR. I wanted to update my fork for my own little project until the next merge.

*17 The value of DELE is random and changes whenever the plugin is saved in the Construction Set for various references with the same base object. The game doesn't seem to care what the value of DELE is, it works all the same. So all DELE now default to 0.

*18 I added the skill decider from the first time but it's less obtuse this time around.

*19 Default is wbFloatDigits.

*20 Its incomplete, mostly just notes.

*21 Only the Oblivion.esm has a non zero value for DELE and the only sensible value for it is FILETIME, which lands in the middle of Oblivion's development. I have not checked it with Version Control in the Construction Set, but I doubt it'll change the value but will follow up on it when I do check.

*22 Whenever a quest no longer has any INFOs in a topic it was associated with, it is removed from the topic. In overrides it is moved to QSTR. This looks to be purely a game editor parameter for filtering dialogue based on quests.

*23 Cannot remember why I did it except it fixed an error when loading. Something about EDID not being initialized yet when loading OBME related forms.

*24 Only a temporary change as it should have a union based on if the form is OBME related.

robertgk2017 committed:
Merge branch 'pr/1418' into Misc

robertgk2017 committed:
Merge Commit

robertgk2017 committed:
Common \ wbWorldMaxHeightData - Exclude FO4 from dfNoCopyAsOverride

robertgk2017 committed:
Script \ Asset Manager - Add handling for .xwm files being valid for records using .wav extension in file paths

robertgk2017 committed:
FO76 \ PLYT - Add ANAM/BNAM names

robertgk2017 committed:
FNV \ WEAP \ DNAM \ Reload Animation - Update Enum/AfterLoad to set any instance of 255 to 0

robertgk2017 committed:
xEdit - change all uses of the name Overlay for Update in relation to the file flag

robertgk2017 committed:
oops forgot some

robertgk2017 committed:
Script \ Export Dialogues - Update 2 Element Paths to reflect recent definition changes

robertgk2017 committed:
TES3 \ GLOB \ AfterLoad - force large negative value check to a Single

robertgk2017 committed:
Script - add gmTES3 as a recognized wbGameMode

robertgk2017 committed:
xEdit \ wbDataFormatNif - Uncomment NIF version check for v10010, resolvesa file exception

robertgk2017 committed:
TES4 \ CREA \ NIFZ - Change name to Model List to make CK

robertgk2017 committed:
And 1 more version + ordered the list

robertgk2017 committed:
Script \ Assets Manager - Add TES3 Support

robertgk2017 committed:
Oops

robertgk2017 committed:
ALL - Group Armor Model sub-records into RStructs

robertgk2017 committed:
Script - Update ARMO/ARMA Model Sub-Record handling

Kuroitsune committed:
Service Flag Update

TES4 does not have a valid service for bit 9, it was a simple mistype of IsTES3 from the original commit.

robertgk2017 committed:
Merge pull request #4 from Kuroitsune/Kuro-Misc

Another Test

robertgk2017 committed:
xeMainForm - Add Yes To All functionality to Change Form Ids

robertgk2017 committed:
Script \ Export Dialogue - Now properly considers Quest Conditions when calculating voice types

robertgk2017 committed:
ALL \ ARMA/ARMO - Add dfStructFirstNotRequired to Armor Model RStructs

robertgk2017 committed:
FO76 \ wbObjectTemplate - fix field name

robertgk2017 committed:
SF1 \ COBJ \ CIFK - Decoded as Instanced Filter Keyword

robertgk2017 committed:
SF1 \ WRLD \ CNAM - Exclude from being set when no Parent Worldspace is set.

robertgk2017 committed:
FO4 \ ReferenceRecord - Add wbRagdoll and XHLT

robertgk2017 committed:
FO4 \ IDLM - Swap Model and QNAM sub-record order

robertgk2017 committed:
FO4 \ LIGH - Swap LNAM and SNAM sub-record order

robertgk2017 committed:
FO4 \ IDLM - Add PNAM/QNAM mutually exclusive sub-records with AfterSets and DontShows

robertgk2017 committed:
FO4 \ INGR - Add PTRN Sub-Record

robertgk2017 committed:
FO4 \ REFR - Add XPWR Sub-Record

jbostrus committed:
Scripting: Add new function RecordFromFileByFormID
  Accepts 2 arguments.
* First is a file, either by text name or as file interface.
* Second is a form ID either as cardinal, integer, or string.

The form ID will be clamped based on the file type, small, medium, full.
The ID will then be adjusted for the file's fully qualified load order ID.

jbostrus committed:
Scripting: Add new function RecordFromFileByHexFormID
  Accepts 2 arguments.
* First is a file, either by text name or as file interface.
* Second is a form ID either as cardinal, integer, or string.

The form ID will be clamped based on the file type, small, medium, full.
The ID will then be adjusted for the file's fully qualified load order ID.

robertgk2017 committed:
Merge branch 'pr/1415' into Misc

robertgk2017 committed:
Revert "Scripting: Add new function RecordFromFileByHexFormID"

This reverts commit c907a9ead352200444b5426119daef0fb66ba5fd.

robertgk2017 committed:
Merge branch 'pr/1415' into Misc

robertgk2017 committed:
Implementation \ TwbContainedInElement.DoAfterSet - Add QUST Children GRUP Handling for FO4+

robertgk2017 committed:
Script \ Assets Manager \ TES5 - Add RACE Tint Mask Textures Support

robertgk2017 committed:
TES5 \ RACE - Update Tint Masks array

robertgk2017 committed:
TES5 \ wbTintLayerToStr - Update Race Tint element paths

robertgk2017 committed:
Initial Oblivion Remastered Support

robertgk2017 committed:
More support for Oblivion Remastered

robertgk2017 committed:
Forgot this

robertgk2017 committed:
Add gmTES4R as a Scripting gamemode check

robertgk2017 committed:
Rebuilt wbTexturedModel again for TES4R

jbostrus committed:
Correct path detections for Oblivion Remaster mode

jbostrus committed:
TES4R: REFR\XACN decoding

robertgk2017 committed:
Merge pull request #5 from jbostrus/Misc

TES4R

robertgk2017 committed:
TES4R \ CELL \ XLRL - Rough structure decoding

robertgk2017 committed:
xeInit \ DetectAppMode - fix TES4R position in GameModes Array

robertgk2017 committed:
TES4R - More definition tweaks to support Oblivion Remastered
